hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1456,7 +1456,12 @@ void BleObject::Observer::notifyScanResultEvent(const hal_ble_scan_result_evt_t&
 }
 
 int BleObject::Observer::processAdvReportEventFromThread(const ble_evt_t* event) {
-    CHECK_TRUE(isScanning_, SYSTEM_ERROR_INVALID_STATE);
+    if (!isScanning_) {
+        if (event->evt.gap_evt.params.adv_report.data.p_data) {
+            BleObject::getInstance().dispatcher()->freeEventData(event->evt.gap_evt.params.adv_report.data.p_data);
```

Comment:
Can we simply jump to `free` label instead to avoid call duplication?

Reply:
Oh, I just realized that in this check we need to return an error, while the `free` will lead to `continue_scanning` and return `SYSTEM_ERROR_NONE`. So we cannot simply jump to the `free` label.